httpx/models.py

```diff
@@ -51,7 +52,7 @@
 
 QueryParamTypes = typing.Union[
     "QueryParams",
-    typing.Mapping[str, PrimitiveData],
+    typing.Mapping[str, typing.Union[PrimitiveData, typing.Sequence[PrimitiveData]]],
```

Comment:
I think we can probably be tighter on this an restrict our expected type to explicitly be `typing.List` rather than genericly be `Sequence`. There's arguments either way, but I reckon we should be okay with forcing a narrower subset where possible.

Certainly that'd be more in line with our `typing.List[typing.Tuple[str, PrimitiveData]]` case below, at any reate.

Reply:
Actually, having had a look at a code example with a tuple usage here, I'm unsure about Sequence vs. List. Maybe already good as it currently is.

Reply:
Yup, allowing for tuples as well as lists was the main reason behind using Sequence here.